packages/core/strapi/lib/services/entity-service/types/result.d.ts

```diff
@@ -0,0 +1,10 @@
+import type { Attribute, Common } from '@strapi/strapi';
+
+type Pagination = { page: number; pageSize: number; pageCount: number; total: number };
+
+export type Result<TSchemaUID extends Common.UID.Schema> = Attribute.GetValues<TSchemaUID>;
+
+export type PaginatedResult<TSchemaUID extends Common.UID.Schema> = {
+  results: Result<TSchemaUID>[];
```

Comment:
```suggestion
  results: (null | Result<TSchemaUID>)[];
```
If given an id that doesn't exist, [entity-manager.load can return null](https://github.com/strapi/strapi/blob/7750c77855085699ff05889211e552f97c3b3cd4/packages/core/database/lib/entity-manager/index.js#L1337)